packages/sdk/src/pubkey.spec.ts

```diff
@@ -0,0 +1,27 @@
+import { Encoding } from "@iov/encoding";
+
+import { encodeSecp256k1Pubkey } from "./pubkey";
+
+const { fromBase64 } = Encoding;
+
+describe("pubkey", () => {
+  describe("encodeSecp256k1Pubkey", () => {
+    it("encodes a full signature", () => {
+      const pubkey = fromBase64("AtQaCqFnshaZQp6rIkvAPyzThvCvXSDO+9AzbxVErqJP");
+      expect(encodeSecp256k1Pubkey(pubkey)).toEqual({
+        type: "tendermint/PubKeySecp256k1",
+        value: "AtQaCqFnshaZQp6rIkvAPyzThvCvXSDO+9AzbxVErqJP",
+      });
+    });
+
+    it("compresses uncompressed public keys", () => {
```

Comment:
nice check... this was to explain to me, right?

Reply:
Last time we talked about signature representation with and without recovery byte. This is about public key representations. Secp256k1 can be expressed as 2x 32 bytes with a `0x04` prefix (uncompressed), e.g.

`041d4c015b00cbd914e280b871d3c6ae2a047ca650d3ecea4b5246bb3036d4d74960b7feb09068164d2b82f1c7df9e95839b29ae38e90d60578b2318a54e108cf8`

Then there is a fancy way to compress this to 32 bytes plus a `0x02` or `0x03` prefix. This compression I never bothered trying to understand. It uses some properties of the curve.

The prefixes indicate the format that is used, which is why the secp256k1 pubkeys are either 65 or 33 bytes long. Ethereum always uses uncompressed, Cosmos always uses compressed.